x-pack/test/api_integration/apis/management/rollup/rollup.js

```diff
@@ -0,0 +1,217 @@
+
+        const { body } = await supertest
+          .get(uri)
+          .expect(200);
+
+        expect(body).to.eql({
+          dateFields: [],
+          keywordFields: [],
+          numericFields: [],
+          doesMatchIndices: false,
+          doesMatchRollupIndices: false,
+        });
+      });
+    });
+
+    describe('Crud', () => {
+      describe('list', () => {
+        it('should return an empty array when there are no jobs', async () => {
+          const { body } = await supertest
+            .get(`${API_BASE_PATH}/jobs`)
+            .expect(200);
+
+          expect(body).to.eql({ jobs: [] });
+        });
+      });
+
+      describe('create', () => {
+        it('should create a rollup job', async () => {
+          const indexName = await createIndexWithMappings();
+          const payload = getJobPayload(indexName);
+
+          return createJob(payload).expect(200);
+        });
+
+        it('should throw a 409 conflict when trying to create 2 jobs with the same id', async () => {
+          const indexName = await createIndexWithMappings();
+          const payload = getJobPayload(indexName);
+
+          await createJob(payload);
+
+          return createJob(payload).expect(409);
+        });
+
+        it('should handle ES errors', async () => {
+          const payload = { job: { id: 'abc', invalid: 'property' } };
+
+          const { body } = await createJob(payload);
+          expect(body.message).to.contain('unknown field [invalid]');
+        });
+
+        it('should list the newly created job', async () => {
+          const indexName = await createIndexWithMappings();
+          const payload = getJobPayload(indexName);
+          await createJob(payload);
+
+          const { body: { jobs } } = await supertest.get(`${API_BASE_PATH}/jobs`);
+          const job = jobs.find(job => job.config.id === payload.job.id);
+
+          expect(job).not.be(undefined);
+          expect(job.config.index_pattern).to.eql(payload.job.index_pattern);
+          expect(job.config.rollup_index).to.eql(payload.job.rollup_index);
+        });
+
+        it('should create the underlying rollup index with the correct aggregations', async () => {
+          const indexName = await createIndexWithMappings();
+          await createJob(getJobPayload(indexName));
+
+          const { body } = await supertest.get(`${API_BASE_PATH}/indices`);
+
+          expect(body[ROLLUP_INDEX_NAME]).to.not.be(undefined);
+
+          // Those are all the aggregations that we defined in the getJobPayload() above
+          const expectedAggregations = [
+            'date_histogram',
+            'max',
+            'min',
+            'terms',
+            'histogram',
+            'avg',
+            'value_count',
+          ];
+
+          const aggregations = Object.keys(body[ROLLUP_INDEX_NAME].aggs);
+          expect(aggregations).to.eql(expectedAggregations);
```

Comment:
What do you think of asserting against the `body` of the response? That way I can understand what we actually expect the response to look like, which I think will be more informative and transparent for me as a reader.

```js
expect(body).to.eql({
  // This way I can see what the body looks like
});
```